### PR TITLE
Major Update

### PR DESCRIPTION
**Pull request by OIVAS7572**

### CODE REVIEW

**Comment by barca-cloud** [comment]
@OIVAS7572  there is some new update in shail choski repo see that!

**Comment by OIVAS7572** [comment]
@barca-cloud 
Don't worry I will surely  update this repo within a week.
Can you  deploy & test this branch?  Is it working

**Comment by barca-cloud** [comment]
> @barca-cloud
> Don't worry I will surely update this repo within a week.
> Can you deploy & test this branch? Is it working

Uhh well i need my computer to be fixed as i cant log in heroku maybe tommorow  i will see

**Comment by barca-cloud** [comment]
And also there us a new thing that makes bot  to draw resign and do etc with api being replcaed :))

**Review by Samanyay2010 — APPROVED** [review]

File: config.py [hunk 1]
```diff
@@ -37,7 +40,7 @@ def load_config(config_file):
             raise Exception("Your config.yml has the default Lichess API token. This is probably wrong.")
 
         if not os.path.isdir(CONFIG["engine"]["dir"]):
-            raise Exception("Your engine directory `{}` is not a directory.".format(CONFIG["engine"]["dir"]))
```
Comment (TheYoBots) [inline config.py:40]: I think it's better to keep this line that remove it.

**Review by TheYoBots — COMMENTED** [review]
Also I think you have made a few additional changes which removed few good commits. I'll fix that tomorrow and send a pull request to this branch. Thanks for updating! Sorry for the late response.

File: config.py (hunk 1 above)
Comment (OIVAS7572) [reply config.py:40]: It changed in ShailChoksi's repo. why they changed

Comment (TheYoBots) [reply config.py:40]: No they didn’t change that in ShailChoksi’s repo, it’s an error fix.

Comment (OIVAS7572) [reply config.py:40]: @TheYoBots
I didn't understand. can you fix this in this branch

Comment (TheYoBots) [reply config.py:40]: Yes I will as soon as I can.

**Comment by TheYoBots** [comment]
I have made all necessary fixes to the `update` branch. Now this repository should be stable and also updated to the latest version of [ShailChoksi/lichess-bot](https://github.com/ShailChoksi/lichess-bot)

**Comment by TheYoBots** [comment]
I have also updated the `variants` branch and the `SfDev` branch to the latest version of the `master` branch.

**Comment by OIVAS7572** [comment]
@TheYoBots
Thank you

But... I didn't set the best values for these lines in config.yml
Is there need of enabling it?
```
  online_moves:
    chessdb_book:
      enabled: false
      min_time: 20
      move_quality: "best"     # One of "all", "good", "best"
      min_depth: 30            # Only for move_quality: "best"
      contribute: true
    lichess_cloud_analysis:
      enabled: false
      min_time: 20
      move_quality: "good"     # One of "good", "best"
      max_score_difference: 50 # Only for move_quality: "good". The maximum score difference (in cp) between the best move and the other moves.
      min_depth: 30
      min_knodes: 0
    online_egtb:
      enabled: false
      min_time: 20
      max_pieces: 7
      source: "lichess"        # One of "lichess", "chessdb"
      move_quality: "best"     # One of "good", "best"
```

**Comment by TheYoBots** [comment]
I thought you didn't want to chnge that. I'll make the necessary changes.

**Comment by TheYoBots** [comment]
Done!

**Pull request merged**